src/MSBuild/XMake.cs

```diff
@@ -626,7 +626,7 @@ string commandLine
 
                     if (!String.IsNullOrEmpty(timerOutputFilename))
                     {
-                        AppendOutputFile(timerOutputFilename, elapsedTime.Milliseconds);
+                        AppendOutputFile(timerOutputFilename, (long)elapsedTime.TotalMilliseconds);
```

Comment:
This is frustratingly easy to do and such a very different result... That's great that it caught this!